astropy/coordinates/representation.py

```diff
@@ -114,9 +119,22 @@ def get_name(cls):
             name = name[:-14]
         return name
 
-    def __getitem__(self, view):
-        return self.__class__(*[getattr(self, component)[view]
-                                for component in self.components])
+    def _replicate(self, method, *args, **kwargs):
+        """Replicate a coordinate object while applying a method to the arrays.
+
+        Parameters
+        ----------
+        method : str
+            The method is applied to the internal ``components``.
```

Comment:
Is it as clear as it could be here that ```method``` must be an attribute of the ```component``` for this to work?